yt/yt/tests/integration/proxies/test_oauth.py

```diff
@@ -35,23 +35,52 @@ def auth_config(port):
                 },
             ],
         },
-        "oauth_cookie_authenticator": {},
-        "oauth_token_authenticator": {},
-        "cypress_user_manager": {},
+        "oauth_cookie_authenticator": {
+            "cache": {
+                "cache_ttl": "5s",
+                "optimistic_cache_ttl": "1m",
+                "error_ttl": "1s",
+            },
+        },
+        "oauth_token_authenticator": {
+            "cache": {
+                "cache_ttl": "5s",
+                "optimistic_cache_ttl": "1m",
+                "error_ttl": "1s",
+            },
+        },
+        "cypress_user_manager": {
+            "cache": {
+                "cache_ttl": "5s",
+                "optimistic_cache_ttl": "1m",
+                "error_ttl": "1s",
+            },
+        },
     }
 
+    if not create_user_if_not_exists:
+        config["oauth_cookie_authenticator"]["create_user_if_not_exists"] = False
+        config["oauth_token_authenticator"]["create_user_if_not_exists"] = False
+
+    return config
+
 
 class TestOAuthBase(YTEnvSetup):
+    NUM_MASTERS = 1
+
     ENABLE_HTTP_PROXY = True
+    NUM_HTTP_PROXIES = 1
     DELTA_PROXY_CONFIG = {}
 
+    OAUTH_CREATE_USER_IF_NOT_EXISTS = True
+
     @classmethod
     def setup_class(cls):
         cls.open_port_iterator = OpenPortIterator(
             port_locks_path=os.environ.get("YT_LOCAL_PORT_LOCKS_PATH", None))
         cls.mock_server_port = next(cls.open_port_iterator)
 
-        cls.DELTA_PROXY_CONFIG["auth"] = auth_config(cls.mock_server_port)
+        cls.DELTA_PROXY_CONFIG["auth"] = auth_config(cls.mock_server_port, cls.OAUTH_CREATE_USER_IF_NOT_EXISTS)
```

Comment:
Let's make it more sophisticated right away, like `DELTA_AUTH_CONFIG` and use `update` here to merge the delta to our default config.